migrations/20231210123554_init.sql

```diff
@@ -0,0 +1,314 @@
+    creator_id  UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (event_id) REFERENCES event (id),
+    FOREIGN KEY (creator_id) REFERENCES event_staff (id),
+    -------------------------------------------------------
+    CONSTRAINT check_event_title_len
+        CHECK (char_length(title) >= 1),
+    CONSTRAINT check_event_stuff_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE assigned_staff
+(
+    status      acceptance_status NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    task_id     UUID,
+    user_id     UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (task_id) REFERENCES task (id),
+    -------------------------------------------------------
+    CONSTRAINT check_assigned_staff_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE comment
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    -------------------------------------------------------
+    content     TEXT NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    event_id    UUID,
+    author_id   UUID,
+    task_id     UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (event_id) REFERENCES event (id),
+    FOREIGN KEY (author_id) REFERENCES "user" (id),
+    FOREIGN KEY (task_id) REFERENCES task (id),
+    -------------------------------------------------------
+    CONSTRAINT check_comment_single_relation_only
+        CHECK (
+            (CASE WHEN event_id IS NULL THEN 0 ELSE 1 END
+             + CASE WHEN task_id IS NULL THEN 0 ELSE 1 END
+            ) = 1),
+    CONSTRAINT check_comment_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+
+);
+
+
+CREATE TABLE associated_company
+(
+    type        association NOT NULL,
+    -------------------------------------------------------
+    created_at  TIMESTAMP NOT NULL DEFAULT now(),
+    edited_at   TIMESTAMP NOT NULL DEFAULT now(),
+    deleted_at  TIMESTAMP,
+    -------------------------------------------------------
+    company_id  UUID,
+    event_id    UUID,
+    -------------------------------------------------------
+    FOREIGN KEY (company_id) REFERENCES company (id),
+    FOREIGN KEY (event_id) REFERENCES event (id),
+    -------------------------------------------------------
+    CONSTRAINT check_associated_company_created_at_lte_edited_at
+        CHECK (edited_at >= created_at)
+);
+
+
+CREATE TABLE adress
+(
+    id          UUID PRIMARY KEY DEFAULT gen_random_uuid(),
```

Comment:
Do we need a UUID here? I understand that it might save a few rows if we use the same address in case two companies are residing at the same address, but even there they might have a slightly different address, and we can just recognize addresses by using the company_id as the PK of this table.